pkg/controller/deployment/deployment_controller.go

```diff
@@ -699,6 +711,28 @@ func (dc *DeploymentController) scaleUpNewRCForRecreate(newRC *api.ReplicationCo
 	return true, err
 }
 
+func (dc *DeploymentController) cleanupOldRcs(oldRCs []*api.ReplicationController, deployment extensions.Deployment) error {
+	diff := len(oldRCs) - *deployment.Spec.RevisionHistoryLimit
+	if diff <= 0 {
+		return nil
+	}
+
+	sort.Sort(controller.ControllersByCreationTimestamp(oldRCs))
+	// TODO: This should be parallelized.
+	for i := 0; i < diff; i++ {
+		controller := oldRCs[i]
+		// Avoid delete rc with non-zero replica counts
+		if controller.Spec.Replicas != 0 || controller.Generation > controller.Status.ObservedGeneration {
+			continue
+		}
+		if err := dc.client.ReplicationControllers(controller.Namespace).Delete(controller.Name); err != nil && errors.IsNotFound(err) {
+			glog.V(2).Infof("Failed deleting old rc %v for deployment %v: %v", controller.Name, deployment.Name, err)
+			return err
```

Comment:
I would not error out here. Maybe try to delete all the controllers, aggregate the errors and return the aggregated error in the end.